### PR TITLE
Add SDK integration doc for inabox.

### PR DESCRIPTION
**Pull request by lannka**
/cc @zombifier (can't add you to reviewer for some reason).

### CODE REVIEW

File: ads/inabox/inabox-host.md [hunk 1]
```diff
@@ -14,4 +14,87 @@ See the License for the specific language governing permissions and
...
+
+###### friendly iframe
+An iframe on the host page that has the same origin, so that it has
+full access to the host page content.
+
+###### foreign iframe
+An iframe on the host page that has different origin, so it can only
+communicate to the host page via postMessage API.
+
+## Web SDK (ads tag) integration
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: I don't think anyone calls this a "web sdk"?

File: ads/inabox/inabox-host.md [hunk 2]
```diff
@@ -14,4 +14,87 @@ See the License for the specific language governing permissions and
...
+Using friendly iframe requires zero to little work for the ads tag 
+integration. Because AMPHTML is trustful & secure, it can be put inside
+ a friendly iframe without worrying it hijacking the page
+and ruining the end user experience. Meantime, friendly iframe rendering
+in general requires less browser resources than foreign iframe rendering,
+which not only brings performance benefit but also revenue gain. One 
+thing to keep in mind though is that the ad network should have their own 
+mechanism to prevent pub
+modifying the ad content or faking clicks. Google Ad Manager uses this
+ approach to integrate with AMPHTML ads.
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: Put something here about the recommended defense-in-depth CSP?

File: ads/inabox/inabox-host.md [hunk 3]
```diff
@@ -14,4 +14,87 @@ See the License for the specific language governing permissions and
...
+
+Main work flow:
+1. The ads tag makes an ad request.
+1. The response contains a flag indicating it is an AMPHTML ad.
+1. The ads tag creates a friendly iframe, and set the response content 
+to the iframe using `srcdoc`.
+
+In the 2nd step, if the ad is generated by a 3rd party, it's a good 
+ idea to run it through an AMP validator to make sure it is valid AMP.
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: In cases where the ad network is relying on AMPHTML being trustworthy, running the ad through the validator is more than a good idea!

File: ads/inabox/inabox-host.md [hunk 4]
```diff
@@ -14,4 +14,87 @@ See the License for the specific language governing permissions and
...
+Main work flow:
+1. The ads tags sets up a postMessage listener on host window.
+1. The ads tag creates an iframe with `src` set to the ad request URL.
+1. If the response is an AMPHTML ad, it will postMessage to the host 
+window. The AMP runtime version number is included in the message.
+1. The ads tag received the message then starts to load the 
+corresponding host script: 
+`https://cdn.ampproject.org/rtv/{version}/amp4ads-host-v0.js`
+1. Before host script is loaded, ads tag is still responsible to queue 
+up all coming AMP messages into a global array.
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: coming -> arriving

File: ads/inabox/inabox-host.md [hunk 5]
```diff
@@ -14,4 +14,87 @@ See the License for the specific language governing permissions and
...
+
+Main work flow:
+1. The ads tags sets up a postMessage listener on host window.
+1. The ads tag creates an iframe with `src` set to the ad request URL.
+1. If the response is an AMPHTML ad, it will postMessage to the host 
+window. The AMP runtime version number is included in the message.
+1. The ads tag received the message then starts to load the 
+corresponding host script: 
+`https://cdn.ampproject.org/rtv/{version}/amp4ads-host-v0.js`
+1. Before host script is loaded, ads tag is still responsible to queue
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: to queue -> for queuing

File: ads/inabox/inabox-host.md [hunk 6]
```diff
@@ -14,4 +14,87 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-This is still in development. Check [here](https://github.com/ampproject/amphtml/issues/5700) for details. 
\ No newline at end of file
+# SDK integration for AMPHTML ad
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: ad -> ads

File: ads/inabox/inabox-host.md [hunk 7]
```diff
@@ -14,4 +14,87 @@ See the License for the specific language governing permissions and
...
+One goal of AMPHTML ad is for advertisers to create ad once and use 
+every where. This guide provides details about how an ad network can 
+join us to serve and render AMPHTML ad in various environment, for 
+example regular web & native mobile apps.
+
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: Mention server-side rendering somewhere?

File: ads/inabox/inabox-host.md (hunk 2 above)
Comment (lannka) [reply ads/inabox/inabox-host.md]: added a `security` session to talk about CSP and validation

File: ads/inabox/inabox-host.md (hunk 3 above)
Comment (lannka) [reply ads/inabox/inabox-host.md]: done

File: ads/inabox/inabox-host.md (hunk 4 above)
Comment (lannka) [reply ads/inabox/inabox-host.md]: done

File: ads/inabox/inabox-host.md (hunk 5 above)
Comment (lannka) [reply ads/inabox/inabox-host.md]: done

File: ads/inabox/inabox-host.md (hunk 1 above)
Comment (lannka) [reply ads/inabox/inabox-host.md]: how about JS SDK?

File: ads/inabox/inabox-host.md (hunk 7 above)
Comment (lannka) [reply ads/inabox/inabox-host.md]: Hesitated to do so... given it's not really standardized at this moment. It makes quite lot assumption with AMP runtime. There is an on-going effort to open-source some SSR for regular AMP doc. But I don't think it's that ready yet, as nothing about amp4ads is covered.

Comment (jeffkaufman) [reply ads/inabox/inabox-host.md]: I think it's worth mentioning that this is in progress, and that if you're looking at serving inabox you should get in touch with the project.  Without SSR performance is pretty bad.

File: ads/inabox/inabox-host.md [hunk 8]
```diff
@@ -14,4 +14,103 @@ See the License for the specific language governing permissions and
...
+the experiment:
+`<meta name="amp-experiments-opt-in" content="inabox-viewport-friendly">`.
+
+###### Security assurance
+In case that the AMPHTML ad is generated by a 3rd party, to ensure it's
+secure to put it in a friendly iframe, the following 2 enforcements 
+should be considered:
+1. Add [Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+ rules to the ad response to 1) only load JS files from trusted AMP 
+ CDN; 2) disallow iframe embed.
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: If you look on a page with GPT you can see Google is currently using:

```
<meta http-equiv="Content-Security-Policy" content="script-src https://cdn.ampproject.org/;object-src 'none';child-src blob:;frame-src 'none'">
```

Why not recommend that here?  If you're not careful, for example, you can break amp-bind by not allowing workers to be loaded from blobs.

See https://github.com/ampproject/amphtml/issues/21681 where I proposed AMP recommend this CSP.

File: ads/inabox/inabox-host.md [hunk 9]
```diff
@@ -14,4 +14,103 @@ See the License for the specific language governing permissions and
...
+1. Validate the AMPHTML ad using an AMP validator. There is an open 
+source JS implementation for the validator. But for performance reason
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: reasons

File: ads/inabox/inabox-host.md [hunk 10]
```diff
@@ -14,4 +14,103 @@ See the License for the specific language governing permissions and
...
+to the iframe using `srcdoc`.
+
+Note that this friendly iframe approach right now is in an experimental
+stage. Put the following meta tag in the head of the ad HTML to opt-in 
+the experiment:
+`<meta name="amp-experiments-opt-in" content="inabox-viewport-friendly">`.
+
+###### Security assurance
+In case that the AMPHTML ad is generated by a 3rd party, to ensure it's
+secure to put it in a friendly iframe, the following 2 enforcements
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: "enforcements" -> "restrictions" or "protections"

File: ads/inabox/inabox-host.md [hunk 11]
```diff
@@ -14,4 +14,103 @@ See the License for the specific language governing permissions and
...
+
+Ideally, both should be implemented for defense-in-depth.
+ 
+##### Cross-domain iframe
+If the ad has to be put into a cross-domain iframe, the ads tag is 
+responsible to set up the communication channel for the ad to complete 
+various tasks such as ad expansion or viewability tracking. A so-called
+host script is to be loaded in that case to fulfill the requirements.
+
+Main work flow:
```
Comment (jeffkaufman) [inline ads/inabox/inabox-host.md]: workflow

File: ads/inabox/inabox-host.md (hunk 7 above)
Comment (lannka) [reply ads/inabox/inabox-host.md]: added a section for `Server side rendering`

**Pull request merged**